packages/react/src/functions/string-to-function.ts

```diff
@@ -118,21 +119,36 @@ export function stringToFunction(
         // browser bundler's like rollup and webpack. Our rollup plugin strips these comments only
         // for the server build
         // TODO: cache these for better performancs with new VmScript
-        // tslint:disable:comment-format
-        const { VM } = safeDynamicRequire('vm2') as typeof import('vm2');
-        const [state, event, _block, _builder, _Device, _update, _Builder, context] = args;
+        const isolateContext: import('isolated-vm').Context = getIsolateContext();
+        const jail = isolateContext.global;
+        // This makes the global object available in the context as `global`. We use `derefInto()` here
+        // because otherwise `global` would actually be a Reference{} object in the new isolate.
+        jail.setSync('global', jail.derefInto());
 
-        return new VM({
-          timeout: 100,
-          sandbox: {
-            ...state,
-            ...{ state },
-            ...{ context },
-            ...{ builder: api },
-            event,
-          },
-        }).run(str.replace(/(^|;)return /, '$1'));
-        // tslint:enable:comment-format
+        // We will create a basic `log` function for the new isolate to use.
+        jail.setSync('log', function (...args: any[]) {
+          if (isDebug()) {
+            console.log(...args);
+          }
+        });
+
+        const ivm = safeDynamicRequire('isolated-vm') as typeof import('isolated-vm');
+        return isolateContext.evalClosureSync(
+          makeFn(str, useReturn),
+          args.map((arg, index) =>
+            typeof arg === 'object'
+              ? new ivm.Reference(
+                  index === 3
```

Comment:
What is the significance of the `3` here? Could be worth a comment or define it as a constant with a descriptive name